indexers/keyvalue/LevelDBIndexer/__init__.py

```diff
@@ -38,7 +38,12 @@ def query(self, keys: Iterable[str], *args, **kwargs) -> Optional[bytes]:
         """
         from google.protobuf.json_format import Parse
 
-        vs = [Parse(self.query_handler.get(bytes(key)).decode('utf8'), Document()) for key in keys]
+        vs = []
+        for key in keys:
+            aux = self.query_handler.get(bytes(key))
+            if aux is not None:
+                vs.append(Parse(aux.decode('utf8'), Document()))
```

Comment:
you need to return `None` in the `list` if the `key` does not `exist`.

Otherwise the client will not be able to know which result correspond to which key.

Reply:
if this test passes, this means the tests are wrong. We need to add a test where we query `present` and `unexisting` keys in the same batch

Reply:
Please fix this and make sure there is a test sensitive to this

Reply:
changed validate_negative_results to take into account the number of elements in the list (not just that missing keys should get None values, but that the number of missing keys should be equal to the number of None values)